backend/src/service/courseService.js

```diff
@@ -501,6 +501,77 @@ const checkOutComeAfterResolve = async (req, res, next) => {
   }
 };
 
+
+// In your courseService.js file
+
+const autoAssignSubject = async (req, res, next) => {
+  try {
+    const userPlanner = req.userPlanner;
+    const planner = userPlanner.planner;
+    const subjectCode = req.params.subjectCode;
+
+    if (!subjectCode) {
+      return res.status(400).json({ error: 'No subject code provided' });
+    }
+
+    // Get the subject from the database
+    const collection = await mongoClient.getCollection('Subject');
```

Comment:
use the constant SUBJECT_COLLECTION instead of hard-coded "Subject" to improve extensibility